flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/taskmanager/AbstractTaskManagerFileHandler.java

```diff
@@ -263,4 +264,10 @@ private void transferFile(ChannelHandlerContext ctx, File file, HttpRequest http
 			throw new FlinkException("Could not transfer file " + file + " to the client.", ioe);
 		}
 	}
+
+	protected String getFileName(HandlerRequest<EmptyRequestBody, M> handlerRequest) {
+		return "undefined";
```

Comment:
Why not just return `null`?

Reply:
Here use the task manager id and file name as the key to LoadingCache.